Fix 1271503, 1274688, 1036150, 712337: Fix profile format and errors.

stephaniehobson opened the pull request:
Fix bug 1271503: Reformat profile.
Moved user links into main profile section. Removed styling for user-photo because it assumed user-links were after it. Reformatted user links to be inline. Some RTL fixes

Fix bug 1274688: JS error on profile.
Included jQuery UI to stop JS errors

Fix bug 1036150: Can't add expertise.
Changed from .attr(checked) to .prop(checked).

Fix bug 712337: Improve expertise.
Show expertise field only if there is expertise for the user to check.

Also includes fix to make error messages red again and to put "change" link inline with the form label it belong to.

schalkneethling commented:
@stephaniehobson I might have uncovered a bug that is not related to this change but also effects the profile. Here is what I am seeing (both locally and on production)

1. I edit my profile and complete the fields for Twitter and Github
2. I click `Publish`
3. The edit profile page reloads and the updated field above are there
4. I click `View profile`
5. Neither of the two above completed fields are visible on my profile
6. I click edit profile again
7. The two above mentioned fields are cleared out.

jwhitlock commented:
@schalkneethling sorry for the long, long time before a response.

The "form error" message on the profile page is almost invisible.  It's in dark grey text below the field, like this:

<img width="657" alt="twitter_error" src="https://cloud.githubusercontent.com/assets/286017/16246485/026a760a-37cb-11e6-8270-90143ed57114.png">

If I hit "Publish", then the "edit profile page" reloads with the easily missable error message. If I enter the expected value (the full Twitter URL https://twitter.com/jwhitlock) and hit publish, the next page is the viewing the profile.

This recently [tripped up a user](https://bugzilla.mozilla.org/show_bug.cgi?id=1276149), so it is possible that something changed. Can you recheck if this is what is happening?

schalkneethling commented:
@jwhitlock Ah, I see the message now but, I get it even when I enter a full url:
![screen shot 2016-06-23 at 14 13 00](https://cloud.githubusercontent.com/assets/10350960/16302653/a4922cca-394c-11e6-9d97-d2bdc9fdec7c.png)

schalkneethling commented:
So, I also see these errors in the console:
![screen shot 2016-06-23 at 14 14 40](https://cloud.githubusercontent.com/assets/10350960/16302712/df9a4e7e-394c-11e6-8e3f-1e65f9f7a92d.png)

jwhitlock commented:
1. Try https://twitter.com/schalkneethling. https://www.twitter.com/schalkneethling will redirect to that address.
2. JS errors are recorded in [bug 1274688](https://bugzilla.mozilla.org/show_bug.cgi?id=1274688), not introduced by this PR
3. MDN is a fractal mess of errors, and a tolerance for imperfection is required to make forward progress

stephaniehobson commented:
New PR, rebased and updated to include: no more javascript errors, error messages are red again.

My tolerance for imperfection is lower than @jwhitlock as evidenced by how much more he gets done ;)

stephaniehobson commented:
@schalkneethling Sorry I was so long responding, there was an all hands and a PTO. The console errors should be gone now and the errors messages are ever so slightly more visible.

schalkneethling commented on kuma/static/js/users.js:
```diff
@@ -60,10 +67,12 @@
         $.each(eTags, function(idx, tag) {
             tag = $.trim(tag);
-            $('#tags-expertise .tag-expert input[value=' + tag + ']').attr('checked', 'checked');
+            if(tag != '') {
```
`!==`?

schalkneethling commented on kuma/static/js/users.js:
```diff
@@ -95,9 +107,13 @@
+        var expertise = $('#id_user-expertise').val().toLowerCase();
+        $('#id_user-expertise').hide()
+        .val(expertise)
+        .after("<ul id='tags-expertise' class='tags'></ul>");
```
Nit: For consistency, should this not be wrapped in single quotes and then double quotes inside?

schalkneethling commented on kuma/users/jinja2/users/includes/user_title_info.html:
```diff
@@ -70,4 +70,38 @@ <h1 class="user-title">
+
+  {% if accounts.github %}
+    {% for github_account in accounts.github %}{% if github_account.extra_data %}
+    <li class="github"><i class="icon-github" aria-hidden="true"></i><a href="{{ github_account.get_profile_url() }}" rel="me external" class="url">{{ _("GitHub") }}</a></li>
```
The `aria-hidden` attributes are not being updated when these items are visible.

stephaniehobson replied:
`aria-hidden` is on the icons, they should always be hidden.

I don't have aria on the list item that is being hidden or shown partially because I didn't know how to hook aria-controls into the the jquery UI widget which is the trigger for showing/hiding this field and partially because I think the logical format of the page is fine from a screen reader perspective. If they have added interests in the preceding step that reveal the expertise options then they will encounter the expertise options naturally when they move to the next step in the form. If something goes wrong and they don't, that's okay, the field isn't required.

schalkneethling commented:
Couple of small items, looking much better. Thx @stephaniehobson

stephaniehobson commented:
Nits picked. 🐞

schalkneethling commented:
@stephaniehobson r+

stephaniehobson commented:
🎉

jwhitlock commented:
Testing in staging. 😍 working Interests, Areas of expertise, field errors. Will be in production soon...

![screenshot 2016-07-11 13 22 07](https://cloud.githubusercontent.com/assets/286017/16741794/837c811c-476a-11e6-8eb0-3b85747dc67d.png)

![screenshot 2016-07-11 13 21 40](https://cloud.githubusercontent.com/assets/286017/16741800/87ef751a-476a-11e6-9841-97ef58d77d11.png)
